api/services/okta_service.py

```diff
@@ -42,7 +44,7 @@ def initialize(self, okta_domain: Optional[str], okta_api_token: Optional[str],
     async def _retry(func: Callable[[Any], Any], *args: Any, **kwargs: Any) -> Any:
         """Retry Okta API requests with specific status codes using exponential backoff."""
         for attempt in range(1 + REQUEST_MAX_RETRIES):
-            result = await func(*args, **kwargs)
+            result =  await asyncio.wait_for(func(*args, **kwargs), timeout=30)
```

Comment:
Let's make the `30` a constant above `REQUEST_TIMEOUT = 30` or something similar